libraries/chain/webassembly/privileged.cpp

```diff
@@ -85,7 +85,7 @@ namespace eosio { namespace chain { namespace webassembly {
       }
       EOS_ASSERT( producers.size() == unique_producers.size(), wasm_execution_error, "duplicate producer name in producer schedule" );
 
-      return context.control.set_proposed_producers( std::move(producers) );
+      return context.control.set_proposed_producers( context.trx_context, std::move(producers) );
```

Comment:
Why not call `context.trx_context.set_proposed_producers()` in the same was as `context.trx_context.set_proposed_finalizers()`?

Reply:
Because `controller` currently is in charge of knowing `legacy` vs `savanna` modes. So route it through `controller` so `controller` can dispatch to correct `set_proposed_producers` or `set_proposed_producers_legacy`.